core/src/saros/account/XMPPAccountStore.java

```diff
@@ -224,31 +200,18 @@ private synchronized void saveAccounts() {
     }
 
     ByteArrayOutputStream out = new ByteArrayOutputStream();
-    ObjectOutputStream oos;
+    XStream xStream = createXStream();
 
-    DataOutputStream dataOut = null;
+    FileOutputStream dataOut = null;
 
     try {
-      dataOut = new DataOutputStream(new FileOutputStream(accountFile));
-
-      oos = new ObjectOutputStream(out);
-      oos.writeObject(activeAccount);
-      oos.flush();
-
-      byte[] activeAccount = Crypto.encrypt(out.toByteArray(), secretKey);
-      out.reset();
-
-      oos = new ObjectOutputStream(out);
-      oos.writeObject(accounts);
-      oos.flush();
+      dataOut = new FileOutputStream(accountFile);
+      // use a pair in order to create a artificial xml root node
+      xStream.toXML(new ImmutablePair(activeAccount, accounts), out);
```

Comment:
List<XMPPAcounts> accountsToSave = new ArrayList<>(accounts)
int activeAccountIdx = accountsToSave.indexOf(acivteAccount)

new ImmutablePair(activeAccountIdx, accountsToSave)

Reply:
I like the idea. This will simplify the XML structure (without a reference)

Reply:
Done